src/main/java/io/jenkins/plugins/sprp/PipelineSnippetGenerator.java

```diff
@@ -0,0 +1,203 @@
+package io.jenkins.plugins.sprp;
+
+import io.jenkins.plugins.sprp.models.Agent;
+import io.jenkins.plugins.sprp.models.ArtifactPublishingConfig;
+import io.jenkins.plugins.sprp.models.Stage;
+import org.apache.commons.lang.StringUtils;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+public class PipelineSnippetGenerator {
```

Comment:
I wonder whether it could be simplified by using a whatever DOM model with further conversion. Generation of Strings does not look fancy with all these tabs. It's also not that good from Performance PoV.